pkg/scheduler/objects/queue.go

```diff
@@ -1659,17 +1659,21 @@ func (sq *Queue) SupportTaskGroup() bool {
 func (sq *Queue) updateGuaranteedResourceMetrics() {
 	if sq.guaranteedResource != nil {
 		for k, v := range sq.guaranteedResource.Resources {
```

Comment:
There's no reason for any of this logic to be here. Encapsulate all of it within pkg/metrics/queue.go.

Reply:
Addressed in latest PR, thanks.